src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -966,5 +1068,106 @@ private static string GetStorageAccountNameFromUriString(string uriStr)
             var index = storageUri.IndexOf('.');
             return storageUri.Substring(0, index);
         }
+
+        private string SshKeyForLinux()
+        {
+
+            if (this.GenerateSshKey.IsPresent && ! this.IsParameterBound(c => c.SshKeyName))
+            {
+                throw new Exception("Please provide parameter '-SshKeyName' to be used with '-GenerateSshKey'");
+            }
+
+
+            // check if -SshKeyName exists in azure
+            Microsoft.Azure.Commands.Compute.Automation.NewAzureSshKey sshKeyClass = new Microsoft.Azure.Commands.Compute.Automation.NewAzureSshKey();
+            SshPublicKeyResource SshPublicKey;
+            string publicKey = "value";
+            try
```

Comment:
logic in line 1085 to line 1139 should be: if(this.GenerateSshKey.IsPresent){ trygetkey() ? exception : generatenewkey } else { trygetkey() ? continue : exception }